Bottom: 7ada0a0d44d6547b22354ce4c6ce7d2f96318663
Top:    8effac3e5768c40d704f996022d8bf6f7c566bd2
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-10-03 23:12:53 +0100

We can just delete kicked out holes

Since we don't keep holes from broken contours, if a contour is
kicked out, we know it is destined for the bin.


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 9da353e..cb70a1c 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1390,7 +1390,8 @@ InsertHoles (jmp_buf * e, POLYAREA * dest, PLINE ** src)
 	}
       else
 	{
-	  /* Need to check if this new hole means we need to kick out any old ones for reprocessing */
+	  /* Need to check if this new hole means we need to kick out any old
+	   * ones for deletion */
 	  while (1)
 	    {
 	      struct find_inside_info info;
@@ -1424,10 +1425,7 @@ InsertHoles (jmp_buf * e, POLYAREA * dest, PLINE ** src)
 
 	      /* Remove hole from the contour */
 	      remove_contour (pa_info->pa, prev, info.result, TRUE);
-
-	      /* Add hole as the next on the list to be processed in this very function */
-	      info.result->next = *src;
-	      *src = info.result;
+	      poly_DelContour (&info.result);
 	    }
 	  /* End check for kicked out holes */
